node/src/bootstrap/bootstrap_attempt.rs

```diff
@@ -109,6 +112,8 @@ impl BootstrapAttempt {
             stopped: AtomicBool::new(false),
             requeued_pulls: AtomicU32::new(0),
             frontiers_received: AtomicBool::new(false),
+            bootstrap_started_observer,
+            bootstrap_ended_observer,
```

Comment:
These two callbacks are always passed together. So I think it would be helpful if we pack these two callbacks into one struct "BootstrapCallbacks" that internally would hold the Arc<Mutex<>>
Then we only need to pass this one instance and if we have to add more callbacks it will only need changes to that new struct.